Document replace tables for Delta Lake and update section for Iceberg (#20819)
trinodb/trino: docs/src/main/sphinx/connector/delta-lake.md

The connector supports the following [](/sql/alter-table) statements.

(delta-lake-create-or-replace)=
#### Replace tables

The connector supports replacing an existing table as an atomic operation.
Atomic table replacement creates a new snapshot with the new table definition as
part of the [table history](#delta-lake-history-table).

To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or
[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).

In this example, a table `example_table` is replaced by a completely new
definition and data from the source table:

```sql
CREATE OR REPLACE TABLE example_table
WITH (partitioned_by = ARRAY['a'])

> AS SELECT * from another_table;

Reviewer: Could you uppercase `from`?. Same for Iceberg example.

Reviewer 2: I fixed that and a few other nits now

The connector supports the following [](/sql/alter-table) statements.

(delta-lake-create-or-replace)=

> 

Reviewer 2: remove empty line

Author: All of the other anchors on the page have a space between the them and the section header.

Reviewer 2: I know .. we want to get rid of them all eventually

### Replace tables

The connector supports replacing an existing table as an atomic operation.
Atomic table replacement creates a new snapshot with the new table definition as
part of the [table history](#delta-lake-history-table).

To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or
[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).

In this example, a table `example_table` is replaced by a completely new

> definition:

Reviewer 2: ```suggestion
definition and data from the source table.
```

```sql
CREATE OR REPLACE TABLE example_table
WITH (partitioned_by = ARRAY['a'])
AS SELECT * from another_table;
```

(delta-lake-alter-table-execute)=

> 

Reviewer 2: remove empty line

- `$file_size`
  : Size of the file for this row.

### Replace tables

The connector supports replacing a table, in case the table already exists, as
an atomic operation. Atomic table replacement creates a new snapshot with the
new table definition and keeps the existing table history. See

> {ref}`delta-lake-history-table` in metadata tables.

Reviewer 2: use markdown syntax for ref link `[](delta-lake-history-table)` .. and optionally with title

To replace a table, use `CREATE OR REPLACE TABLE` and/or `CREATE OR REPLACE

> TABLE AS`. For more information on creating tables, see {doc}`CREATE TABLE

Reviewer 2: use markdown syntax for link
```
[](/sql/create-table)
```
same for others

</sql/create-table>` and {doc}`CREATE TABLE AS </sql/create-table-as>`. These
statements do not support changing table formats defined by the `type` table
property.

After replacement, the table definition is completely new and separate from the
old table.

For example, a table `example_table` can be replaced by a completely new
definition:

```
CREATE TABLE example_table (
    a BIGINT,
    b DATE,
    c BIGINT)

> WITH (partitioning = ARRAY['a']);

Reviewer 2: confirm this is valid table property in delta lake

Reviewer: It should be `partitioned_by`. 
https://github.com/trinodb/trino/blob/23e5ec5b3d30182548d90636a8fbaec9c82ffcfe/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java#L41

CREATE OR REPLACE TABLE example_table

> WITH (sorted_by = ARRAY['a'])

Reviewer 2: confirm table property

Reviewer: The connector doesn't support such property.

- `$file_size`
  : Size of the file for this row.

### Replace tables

> The connector supports replacing a table, in case the table already exists, as

Reviewer 2: ```suggestion
The connector supports replacing an existing table as
```

an atomic operation. Atomic table replacement creates a new snapshot with the

> new table definition and keeps the existing table history. See

Reviewer 2: ```suggestion
new table definition as part of the [table history](#delta-lake-history-table).
```

> [](#delta-lake-history-table) in metadata tables.

Reviewer 2: remove

> To replace a table, use `CREATE OR REPLACE TABLE` and/or `CREATE OR REPLACE

Reviewer 2: ```suggestion
To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or [`CREATE OR REPLACE AS`](/sql/create-table-as).
```

> TABLE AS`. For more information on creating tables, see [](/sql/create-table)

Reviewer 2: Remove from "For more information" onwards

and [](/sql/create-table-as). These statements do not support changing table

> formats defined by the `type` table property.

Reviewer 2: remove this . there is no type table property in Trino.

> After replacement, the table definition is completely new and separate from the

Reviewer 2: Remove .. this is wrong

old table.

For example, a table `example_table` can be replaced by a completely new
definition:

> ```

Reviewer 2: ```suggestion
```sql
```

CREATE TABLE example_table (
    a BIGINT,
    b DATE,
    c BIGINT)
WITH (partitioning = "partitioned_by");

> CREATE OR REPLACE TABLE example_table

Reviewer 2: move this to separate code block and have intro sentence that this create new table with data

- `$file_size`
  : Size of the file for this row.

> ### Replace tables

Reviewer: How about moving to before `ALTER TABLE EXECUTE`?

Reviewer 2: Sure .. that works too.

> The connector supports replacing an existing table, as

Reviewer 2: I think no comma 

```suggestion
The connector supports replacing an existing table as
```

Reviewer 2: Also fix wrapping of this paragraph

an atomic operation. Atomic table replacement creates a new snapshot with the
new table definition as part of the [table history](#delta-lake-history-table).

To replace a table, use [`CREATE OR REPLACE TABLE`](/sql/create-table) or

> [`CREATE OR REPLACE AS`](/sql/create-table-as).

Reviewer 2: ```suggestion
[`CREATE OR REPLACE TABLE AS`](/sql/create-table-as).
```

> In this example, a table `example_table` can be replaced by a completely new

Reviewer 2: ```suggestion
In this example, a table `example_table` is replaced by a completely new
```

definition:

```sql
CREATE TABLE example_table (
    a BIGINT,
    b DATE,
    c BIGINT)

> WITH (partitioning = "partitioned_by");

Reviewer 2: that has to be a column of the table or a function to process the table. See other example

Author: @ebyhr?

Reviewer 2: Either same as other example or just remove it. Having that partitioning defined does nothing to the value of the example

```

In this example, a new table is created with data:

```sql
CREATE OR REPLACE TABLE example_table
WITH (partitioned_by = ARRAY['a'])
AS SELECT * from another_table;
(delta-lake-fte-support)=

> ```

Reviewer: This line should be before `(delta-lake-fte-support)=`

Reviewer 2: yes.. might be github diff issue or actually wrong

Author: This is just weird.
